wordpress-mobile/WordPress-Android
# Site block icon interactions

Fixes #13343 

This PR adds interactions to the new site icon. The dialog has 3 options - change icon, remove icon and dismiss. Change icon should open the media picker and correctly handle the selected icon. The remove icon should start showing the placeholder and dismiss just hides the dialog. I've had to use extract quite a lot of code for this. I've created a new view model to handle the upload success/error events. 

To test:
- Click on the site icon
- Select "CHANGE"
- Notice the media picker is opened
- Select an image from the "Choose from device" option
- Notice the image cropper is opened
- Crop your icon and confirm
- The site icon upload starts
- The site icon progress bar shows
- After the site icon upload finishes successfully the icon is updated and the progress bar is hidden
- Test this with the other media picker options - "Take a photo", "WP Media library" and "System picker"

To test 2:
- Click on the site icon
- Select "REMOVE"
- Notice the progress bar is shown for a short while
- Notice the icon gets removed and the placeholder is visible
- Click on the site icon again
- Notice the dialog is now different with only 2 options


PR submission checklist:
